xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1455,10 +1454,20 @@ void CGUIMediaWindow::OnInitWindow()
 
   m_rootDir.SetAllowThreads(true);
 
-  if (m_iSelectedItem > -1)
-    m_viewControl.SetSelectedItem(m_iSelectedItem);
-
   CGUIWindow::OnInitWindow();
+
+  m_viewControl.SetSelectedItem(GetMediaSelectedItem(m_vecItems->GetPath()));
+}
+
+int CGUIMediaWindow::GetMediaSelectedItem(const std::string& strDirectory)
+{
+  auto it = m_mediaSelectedItems.find(strDirectory);
+  return (it != m_mediaSelectedItems.end()) ? it->second : 0;
```

Comment:
Can you explain the condition? What case is this covering? I struggle to understand it.
Btw the braces shouldn't be needed here.